src/controllers/resource-store/utils/generate-time-guard/index.ts

```diff
@@ -0,0 +1,21 @@
+import { GenerateTimeGuardReturn } from './types';
+
+export const generateTimeGuard = (
```

Comment:
Can we add two specific unit tests for this funciton, just to ensure it works as expected (returned loading state and resolved state)